### PR TITLE
openshift-installer-presubmits: Run go-lint on cmd/... and pkg/...

### PR DESCRIPTION
**Pull request by wking**
Cover these directories, which are new with openshift/installer#120.

### CODE REVIEW

File: ci-operator/config/openshift/installer/master.json [hunk 1]
```diff
@@ -41,7 +41,7 @@
     {
       "as": "unit",
       "from": "src",
-      "commands": "go test ./installer/pkg/..."
+      "commands": "go test ./pkg/... ./installer/pkg/..."
```
Comment (stevekuznetsov) [inline ci-operator/config/openshift/installer/master.json]: Why not `./...` ?

File: ci-operator/jobs/openshift/installer/openshift-installer-presubmits.yaml [hunk 2]
```diff
@@ -293,5 +293,7 @@ presubmits:
         - sh
         args:
         - ./hack/go-lint.sh
+        - cmd/...
```
Comment (stevekuznetsov) [inline ci-operator/jobs/openshift/installer/openshift-installer-presubmits.yaml]: Why not `go list -f '{{ .Dir }}' ./...` ?

File: ci-operator/config/openshift/installer/master.json (hunk 1 above)
Comment (wking) [reply ci-operator/config/openshift/installer/master.json]: > Why not `./...` ?

We are currently leaning on Go's test framework for [smoke tests][1].  We may drop those soon in favor of more e2e-aws coverage (#1271 is groundwork for this expansion), but until then we need to keep this job out of `tests/smoke`.  A white-list like this is one way to do that.  There may also be blacklist approaches, but I think we'd need to include `find` in our invocation if we wanted to do that.

[1]: https://github.com/openshift/installer/tree/245fc0f6313697ecbb2732c4f7184aebc8f0be52/tests/smoke

File: ci-operator/jobs/openshift/installer/openshift-installer-presubmits.yaml (hunk 2 above)
Comment (wking) [reply ci-operator/jobs/openshift/installer/openshift-installer-presubmits.yaml]: > Why not `go list -f '{{ .Dir }}' ./...` ?

Something like that could work, although:

```console
$ sudo hack/go-lint.sh $(go list -f '{{ .Dir }}' ./...)
import "/home/trking/.local/lib/go/src/github.com/openshift/installer/cmd/openshift-install": cannot import absolute path
...
```

It works with `.ImportPath`:

```console
$ sudo hack/go-lint.sh $(go list -f '{{ .ImportPath }}' ./...)
```

How do I encode that in YAML here?  Something like:

```yaml
      - image: quay.io/coreos/golang-testing
        env:
        - name: IS_CONTAINER
          value: "TRUE"
        command:
        - sh
        args:
        - -c
        - ./hack/go-lint.sh $(go list -f '{{ .ImportPath }}' ./...)
```

Comment (stevekuznetsov) [reply ci-operator/jobs/openshift/installer/openshift-installer-presubmits.yaml]: I suspect that PodSpec would work, yes

Comment (wking) [reply ci-operator/jobs/openshift/installer/openshift-installer-presubmits.yaml]: > I suspect that PodSpec would work, yes

Updated with a69bc01 -> aaa039c.

**Comment by stevekuznetsov** [comment]
/lgtm
Will need a rebase

**Comment by wking** [comment]
Rebased around #1265 with aaa039c -> 8f82768.

**Pull request merged**